Improve drifted URLs

janekdb opened the pull request:
:star: Any.scala: Link to the guide instead of the SIP.
:star: AnyVal.scala: Remove SIP link and align guide link to Any.scala.
:star: Commands.scala: Use a less out of date team link.
:star: Logic.scala: Link was broken. Substitute found.
:star: Process.scala: Links were 403 & 404. Fixed as this is a code sample.
:star: TypeMaps.scala: Move old EPFL Trac to JIRA.
:star: RedBlackTree.scala: Replaced broken link with substitutes based on site maintainer input :one:.

:one: When asked where `Data-Set-RBTree.html` had gone Don@UNSW advised

```
 "I think it's on the Haskell wiki now. It was Chris Okazaki's version".
```

The closest I could find to what this document probably was is this paper by Hinze edited by Okasaki,

```
  http://www.cs.ox.ac.uk/ralf.hinze/publications/WAAAPL99b.ps.gz
```

The paper cites the Okasaki document so I included a link to that as well.

The Haskell Wiki does have a link to a RB document but that's broken too,

```
  404: https://wiki.haskell.org/Research_papers/Data_structures > Constructing red-black trees
```

:v:

SethTisue commented on src/library-aux/scala/Any.scala:
```diff
@@ -27,7 +27,7 @@ package scala
  * }}}
  *
- * See the [[http://docs.scala-lang.org/sips/completed/value-classes.html value classes guide]] for more
+ * See the [[http://docs.scala-lang.org/sips/completed/value-classes.html SIP-15: Value Classes]] for more
```
better to link to http://docs.scala-lang.org/overviews/core/value-classes.html here, actually, IMO

janekdb replied:
:heavy_check_mark: Done.

janekdb committed:
Improve drifted URLs

 - Any.scala: Link to the guide instead of the SIP.
 - AnyVal.scala: Remove SIP link and align guide link to Any.scala.
 - Commands.scala: Use a less out of date team link.
 - Logic.scala: Link was broken. Substitute found.
 - Process.scala: Links were 403 & 404. Fixed as this is a code sample.
 - TypeMaps.scala: Move old EPFL Trac to JIRA.
 - RedBlackTree.scala: Replaced broken link with substitutes based on site maintainer input [1].

[1] When asked where Data-Set-RBTree.html had gone Don@UNSW advised
      "I think it's on the Haskell wiki now. It was Chris Okazaki's version".
    The closest I could find to what this document probably was is this
    paper by Hinze edited by Okasaki,
      http://www.cs.ox.ac.uk/ralf.hinze/publications/WAAAPL99b.ps.gz
    The paper cites the Okasaki document so I included a link to that as
    well.
    The Haskell Wiki does have a link to a RB document but that's broken
    too,
      https://wiki.haskell.org/Research_papers/Data_structures >
        Constructing red-black trees

janekdb commented:
Review by @heathermiller, @dickwall, @som-snytt.

soc commented:
LGTM.

SethTisue commented:
good stuff as usual, thanks Janek
